packages/web-app/src/components/appli/Entry/Snapshots/index.jsx

```diff
@@ -0,0 +1,188 @@
+  }
+}
+
+const SnapshotPage = () => {
+  const { formatMessage } = useIntl();
+  const dispatch = useDispatch();
+  const location = useLocation();
+  const queryParameters = new URLSearchParams(location.search);
+  const isNetwork = queryParameters.get('isNetwork') === 'true';
+  const [actualTItem, setActualTItem] = useState({});
+
+  useEffect(() => {
+    const json = localStorage.getItem('t_item');
+    const items = json !== 'undefined' ? JSON.parse(json) : undefined;
+    if (items) {
+      setActualTItem(items);
+    }
+    return localStorage.removeItem('t_item');
+  }, []);
+
+  const { id, type } = useParams();
+
+  const { data, error, status, latestHttpCode } = useSelector(
+    state => state.snapshots
+  );
+
+  useEffect(() => {
+    dispatch(fetchSnapshot(id, type, isNetwork));
+  }, [id, type, isNetwork, dispatch]);
+
+  const isLoading = status === REDUCER_STATUS.LOADING;
+  const isSuccess = status === REDUCER_STATUS.SUCCEEDED;
+  const is404 = error && latestHttpCode === 404;
+  const is403 = error && latestHttpCode === 403;
+  const isSensitive = pathOr(false, ['entrances', '0', 'isSensitive'], data);
+  return (
+    <>
+      {isSensitive && <SensitiveCaveWarning />}
+      {Object.keys(actualTItem).length > 0 && (
+        <ScrollableContent
+          dense
+          title={formatMessage({ id: `Actual ${type}` })}
+          content={
+            <>
+              {type !== 'riggings' && (
+                <Typography variant="h4">
+                  {actualTItem.title ?? actualTItem.name}
+                </Typography>
+              )}
+              {renderComponent(type, actualTItem, isNetwork ?? false)}
+              <Contribution
+                author={getAuthor(actualTItem.author)}
+                creationDate={actualTItem.date}
+                reviewer={getAuthor(actualTItem.reviewer)}
+                dateReviewed={actualTItem.dateReviewed}
+              />
+            </>
+          }
+        />
+      )}
+      {is403 && (
+        <Alert
+          title={formatMessage({
+            id: `This ${type} is related to a sensitive entrance. Only administrator can see this.`
+          })}
+          severity="warning"
+        />
+      )}
+      {is404 && (
+        <Alert
+          title={formatMessage({
+            id: `This is the only version of this ${type}.`
+          })}
+          severity="info"
+        />
+      )}
+      {isLoading && <Skeleton height={300} />}
+      {isSuccess && (
+        <ScrollableContent
+          dense
+          title={formatMessage({ id: `${type} snapshots` })}
+          content={
+            data && Object.keys(data).length > 0 ? (
+              Object.keys(data).map(snapshotType =>
+                data[snapshotType].map(snapshot => {
+                  const author = getAuthor(snapshot.author);
+                  const reviewer = getAuthor(snapshot.reviewer);
+                  return (
+                    <Accordion>
```

Comment:
We could separate this into a dedicated component.

It's better to add a key when mapping:
https://beta.reactjs.org/learn/rendering-lists#keeping-list-items-in-order-with-key